internal/ethapi/api.go

```diff
@@ -1475,9 +1512,45 @@ func (s *PublicTransactionPoolAPI) GetTransactionReceipt(ctx context.Context, ha
 	if receipt.ContractAddress != (common.Address{}) {
 		fields["contractAddress"] = receipt.ContractAddress
 	}
+	if authToken, ok := s.b.SupportsMultitenancy(ctx); ok {
+		extraDataReader, err := s.b.AccountExtraDataStateReaderByNumber(ctx, rpc.BlockNumber(blockNumber))
+		if err != nil {
+			return nil, fmt.Errorf("no account extra data reader at block %v: %w", blockNumber, err)
+		}
+
+		filteredLogs := make([]*types.Log, 0)
+		for _, l := range receipt.Logs {
+			ok, err := s.hasAccessToContract(ctx, authToken, extraDataReader, l.Address)
+			if err != nil {
+				return nil, err
+			}
+			if ok {
+				filteredLogs = append(filteredLogs, l)
+			}
+		}
+		fields["logs"] = filteredLogs
+		receiptClone := &types.Receipt{PostState: receipt.PostState, Status: receipt.Status, Logs: filteredLogs}
+		fields["logsBloom"] = types.CreateBloom(types.Receipts{receiptClone})
+	}
```

Comment:
2. Without multitenancy, `contractAddress` is always visible even if user was not a participant.
1. and 3. seem having the same context, do you foresee any further security concern after knowing if a contract is accessible or not? I would argue to keep the original tx status to reflect the correct status of the transaction execution. Without multitenancy, party-protection transactions also produce fail status for transactions sent by non-party.

Reply:
2. Without multitenancy, a non participant would see an empty `contractAddress` and not the effective deployment address (since a non-participant has no access to the tx payload for deployment). Or am I missing something here?

1. 3. In a non-multitenant world, a non-participant always sees a success status for private transactions. Or am I missing something here?